python-avd/tests/pyavd/molecule_scenarios/test_get_device_test_catalog.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2025 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+import json
+from copy import deepcopy
+from typing import Any
+
+import pytest
+
+from pyavd import get_device_test_catalog
+from pyavd._anta.lib import AntaCatalog
+from pyavd.api._anta import AvdCatalogGenerationSettings, InputFactorySettings, get_minimal_structured_configs
+from tests.models import MoleculeHost, MoleculeScenario
+
+
+# Helper Function for Filter Logic
+def _get_filtered_settings_for_host(molecule_host: MoleculeHost) -> AvdCatalogGenerationSettings:
+    """Replicates the Ansible plugin logic to get the correct filtered settings for a host."""
+    # Filter rules are defined here, mirroring the playbook converge file.
+    avd_catalogs_filters: list[dict[str, Any]] = [
+        {"skip_tests": ["VerifyNTP"]},
+        {"device_list_group": "DC1_SVC_LEAVES", "run_tests": ["VerifyReachability"]},
+        {"device_list_group": "DC2_SPINES", "run_tests": ["VerifyLLDPNeighbors"], "skip_tests": ["VerifyLLDPNeighbors"]},
+    ]
+
+    host_groups = molecule_host.hostvars.get("group_names", [])
+    final_filters: dict[str, Any] = {}
+
+    # Apply filters based on the host's group membership.
+    for filter_config in avd_catalogs_filters:
+        device_list_group = filter_config.get("device_list_group")
+        if device_list_group and device_list_group not in host_groups:
+            continue
+
+        # "Last filter wins" logic updates the dictionary.
+        if "run_tests" in filter_config:
+            final_filters["run_tests"] = filter_config["run_tests"]
+        if "skip_tests" in filter_config:
+            final_filters["skip_tests"] = filter_config["skip_tests"]
+
+    return AvdCatalogGenerationSettings(**final_filters)
+
+
+@pytest.mark.molecule_scenarios("anta_runner")
+@pytest.mark.parametrize(
+    ("run_name", "expected_catalog_property_name"),
+    [
+        ("default", "default_test_catalog"),
+        ("allow_bgp_vrfs", "allow_bgp_vrfs_test_catalog"),
+        ("filtered", "filtered_test_catalog"),
+    ],
+    ids=["default_run", "allow_bgp_vrfs_run", "filtered_run"],
+)
```

Comment:
If we decide to have a single `get_test_catalog` method:
```suggestion
@pytest.mark.parametrize(
    ("run_name"),
    [
        ("default_run"),
        ("allow_bgp_vrfs_run"),
        ("filtered_run"),
    ],
    ids=["default_run", "allow_bgp_vrfs_run", "filtered_run"],
)
```

Reply:
updated